src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1590,6 +1623,108 @@ internal void ParseLinkHeader(HttpResponseMessage response, System.Uri requestUr
             }
         }
 
+        /// <summary>
+        /// Adds content to a <see cref="MultipartFormDataContent" />. Object type detection is used to determine if the value is String, File, or Collection.
+        /// </summary>
+        /// <param name="fieldName">The Field Name to use.</param>
+        /// <param name="fieldValue">The Field Value to use.</param>
+        /// <param name="formData">The <see cref="MultipartFormDataContent" />> to update.</param>
+        /// <param name="enumerate">If true, collection types in <paramref name="fieldValue" /> will be enumerated. If false, collections will be treated as single value.</param>
+        private void AddMultipartContent(object fieldName, object fieldValue, MultipartFormDataContent formData, bool enumerate)
+        {
+            if (null == formData)
```

Comment:
The pattern is common in the file.

Reply:
FYI, this type of null check (null on the LHS) is a C++ idiom that does not apply to C# because if/while/do conditions in C# must evaluate to a Boolean.  Therefore a mistaken assignment e.g. `if (formData = null)` would fail to compile because the type of formData can't be a Boolean (not if you're assigning null to it).  In fact, the only time you need to put the literal on the LHS is when the variable type is a Boolean e.g. `if (false == aBooleanVariable)`.  

My team's internal coding guidelines suggest not using `==` or `!=` in the Boolean variable case but use `if (!aBooleanVariable)` or `if (aBooleanVariable)` instead.  Consequently, we never put the literal value on the LHS in C# code.  And IMO this approach reads easier e.g. `if (formData == null)`.

Reply:
@rkeithhill  The problem is more that this file uses inconsistent forms and I forget which one is being used in which context. In some parts of the file null is left in others null is right. As long as there isn't some programmatic consideration (I suspected there wasn't), I don't have an opinion either way. `null == foo` and `foo == null` read exactly the same to me with out one being easier than the other. I have a habit of "literals go left" because it is a common pattern in many languages for various reasons.

Reply:
@SteveL-MSFT I'm not entire certain of the future of these new methods. I get the feeling they may end up moving to a helper class as static methods to be consumed in more places. throw seems more portable, IMO.

Reply:
I like `(formData == null)` too.
But here we should use the file pattern.

Reply:
@markekraus since the pattern already exists in this file, I'm fine keeping it for now.  Consider this closed for me.